test/upgrade/installation/openshift.go

```diff
@@ -88,3 +94,78 @@ func IsClusterVersionWithImageReady(image string) ClusterVersionInStateFunc {
 		return false
 	}
 }
+
+func IsChannelEUS(ctx *test.Context) (bool, error) {
+	clusterVersion, err := ctx.Clients.ConfigClient.ClusterVersions().Get(context.Background(), clusterVersionName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+	return strings.Contains(clusterVersion.Spec.Channel, "eus"), nil
+}
+
+// UpgradeEUS follows guidelines from https://docs.openshift.com/container-platform/4.8/updating/preparing-eus-eus-upgrade.html
+// to upgrade OpenShift between two EUS versions.
+func UpgradeEUS(ctx *test.Context) error {
+	if updated, err := allMachineConfigPoolsUpdated(ctx); err != nil || !updated {
+		return fmt.Errorf("unable to proceed with upgrades: %w", err)
+	}
+
+	pauseMachineConfigPool(ctx, true)
+
+	if err := UpgradeOpenShift(ctx); err != nil {
+		return fmt.Errorf("failed to upgrade to odd OpenShift release: %w", err)
+	}
+
+	if err := UpgradeOpenShift(ctx); err != nil {
+		return fmt.Errorf("failed to upgrade to even OpenShift release: %w", err)
+	}
```

Comment:
Why is this done twice?

Reply:
OpenShift cannot skip a version during upgrades. It needs to go through each upgrade one-by-one. It's also described in the docs I linked.

Reply:
So, the `UpgradeOpenShift` func upgrade OCP to next major version. I see.

How about we make it clear in the func name: `UpgradeOpenShiftToNextMajor`?

Reply:
It's not exactly "next major". It upgrades to the last version (e.g. 4.9.54) from the next stream 4.9 and then to the last version (4.10.45) of the next stream 4.10. Not sure how to better name it than UpgradeEUS. I also included this comment:
```
// UpgradeEUS follows guidelines from https://docs.openshift.com/container-platform/4.8/updating/preparing-eus-eus-upgrade.html
// to upgrade OpenShift between two EUS versions.
```
maybe "UpgradeOpenShiftEUSToEUS` ?

Reply:
I was thinking about renaming the already present `UpgradeOpenShift` func. but, maybe that would be obvious if the code that uses it be a bit clearer about the intent.

How about we make it clear and at the same time dynamic. Now, you are assuming the EUS to EUS is 2 releases apart. OCP could switch to have longer window in the future.

How about:

```golang
for upgraded := false; !upgraded; upgraded = isEUSRelease(ctx) {
  if err := UpgradeOpenShift(ctx); err != nil {
    return fmt.Errorf("failed to upgrade to OpenShift release: %w", err)
  }
}
```

Reply:
Hmm. There's no easy/reliable way to figure out if any release is EUS. The only way is to compare the MAJOR.MINOR version of CSV with the name of the channel (e.g. eus-4.10). And that sounds easier to change than changing the whole strategy by OpenShift team of having EUS for every second minor version of OCP.

FWIW, I'm not assuming anything. This is the approach that is described in [docs](https://docs.openshift.com/container-platform/4.10/updating/preparing-eus-eus-upgrade.html#updating-eus-to-eus-upgrade_eus-to-eus-upgrade) for EUS-to-EUS upgrade and it is clearly stated that every "all even-numbered minor versions of OpenShift Container Platform offer Extended Update Support (EUS)"

I know you're seeing a little duplication in the code but it is just one line (even the error message is different in this case). Using a loop only makes it more confusing by suggesting that there can be any number of openshift upgrades.

Reply:
Okay, I get you.

Could you please just add a comment like:

```golang
// Calling UpgradeOpenShift twice to upgrade from one EUS release to another EUS release
```

Reply:
Done.

Reply:
BTW. I had proposed that loop because I saw there was this function already: `installation.IsChannelEUS(ctx)`.